fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/planning/InExtractor.java

```diff
@@ -281,6 +242,78 @@ public List<InSource> unionSources() {
         return inClauses.stream().map(InClause::unionSource).collect(Collectors.toList());
     }
 
+    public InExtractor filter(@Nonnull Predicate<String> inBindingFilter) {
+        return fromFilter(this.filter, inBindingFilter);
+    }
+
+    @Nonnull
+    @SuppressWarnings("unchecked")
+    public static InExtractor fromFilter(@Nonnull final QueryComponent filter, @Nonnull Predicate<String> inBindingFilter) {
+        final AtomicInteger bindingIndex = new AtomicInteger();
+        final List<InClause> inClauses = Lists.newArrayList();
+        final QueryComponent subFilter = mapClauses(filter, (withComparison, fields) -> {
+            if (withComparison.getComparison().getType() == Comparisons.Type.IN) {
+                final String bindingName = Bindings.Internal.IN.bindingName(
+                        withComparison.getName() + "__" + bindingIndex.getAndIncrement());
```

Comment:
Are these binding names stable enough? In particular, I think the requirement here is that if `inBindingFilter.test(bindingName)` is `false`, then skipping the logic below must not change any of the in-bindings, if I understand how this is used.

The only alternative I could think of would be to have the `InExtractor` run once to generate the transformed query, and then when we want to turn off certain `IN`-joins, we have to have the `InExtractor` replace the set of clauses with their original comparisons. But in this alternative reality, it doesn't re-run the assignment of clauses to binding names.